src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -134,7 +134,7 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
             if (!skipManifestUpdate)
             {
                 var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
-                if (File.Exists(installStateFilePath))
```

Comment:
I'm wondering why we don't move line 160 up to 132 and just use "!useRollback" here? Seems like the code may have been trying to save the isnull check but I can't imagine that's a slow check.

Reply:
Yeah; I can do that. I typically try to change as little as possible for servicing changes, but that shouldn't hurt anything.

Reply:
Just take note that on line 168, the default `useRollback` value is passed through and may not have been set based on if the `skipManifestUpdate` option that controls the conditional block on line 145.

Reply:
joeloff is right, and in particular, deciding to skip the manifest update isn't decided yet on line 132.

The useRollback parameter passed into InstallWorkloadsWithInstallRecord is only used to modify the install state. If we legitimately rolled back to a specific set of manifests, it is created with those manifests. If not, it is deleted (in 8.0.1xx). Moving useRollback up may make it create the install state when it didn't update manifests at all, much less from a rollback file.